sw/device/lib/crypto/aes_gcm/aes_gcm.c

```diff
@@ -399,16 +399,20 @@ aes_error_t aes_gcm_ghash(const aes_block_t *hash_subkey,
 }
 
 /**
- * One-shot version of the AES API for a single block, for convenience.
+ * One-shot version of the AES encryption API for a single block.
  */
-static aes_error_t aes_single_block(const aes_params_t params,
-                                    const aes_block_t *input,
-                                    aes_block_t *output) {
-  aes_error_t err = aes_begin(params);
+static aes_error_t aes_encrypt_block(const aes_key_t key, const uint32_t iv[4],
```

Comment:
re. `const uint32_t iv[4]`. 

It is recommended that for parameters that are exposed in the public interface, to wrap them inside a struct for type safety. I think it is ok to leave as is in this case, but @alphan can confirm.